crates/sui-types/src/sui_system_state.rs

```diff
@@ -553,6 +581,104 @@ impl Default for SuiSystemState {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, JsonSchema)]
+pub enum SuiSystemState {
+    V1(SuiSystemStateInnerV1),
+}
+
+impl SuiSystemState {
+    /// Always return the initial (genesis) version of the system state object.
+    /// Genesis always uses this version regardless of the current version.
+    pub fn into_genesis_version(self) -> SuiSystemStateInnerV1 {
+        match self {
+            SuiSystemState::V1(inner) => inner,
```

Comment:
This is not really a good assumption. We may start perform genesis after there are many versions (e.g. if starting a new testnet), and we will presumably want to use the most recent version.

If you want to take protocol version as an analogy, we always use the highest supported version for genesis.

Reply:
I don't quite get it. Why do we need to keep multiple versions in the code when we perform genesis?